azure-pipelines.yml

```diff
@@ -1,11 +1,96 @@
-# Starter pipeline
-# Start with a minimal pipeline that you can customize to build and deploy your code.
-# Add steps that build, run tests, deploy, and more:
 # https://aka.ms/yaml
+variables:
+  # TODO: Set this where necessary
+  travis_event_type: ''
```

Comment:
Is this because of the various places in our scripts that assume it? I'd probably rather we refactor the scripts than set this variable in azure!

Reply:
(I'm not suggesting you do that as part of this PR of course)

Reply:
Yeah, this is a quick placeholder because the [hypothesistooling/__main__.py](https://github.com/HypothesisWorks/hypothesis/blob/ccc1ccf8f8c132e09ba397606a817ea3be807692/tooling/src/hypothesistooling/__main__.py#L263) directly indexes os.environ for `TRAVIS_EVENT_TYPE`. 

I can go ahead and fix that there, but I think I'd prefer this to be prominent in the short term while I'm still learning the tooling codebase so that when I go in to flesh the deploy out I remember to, for example, ensure PRs skip deployment steps.

Reply:
Yeah, that makes sense. I'd appreciate a bit more of an explanatory comment here, but don't feel you need to change any of the code.

Reply:
Will add that in the next commit